### PR TITLE
Add ability to remove institutions

### PR DESCRIPTION
**Pull request by danielkryska**
image

### CODE REVIEW

**Comment by Marcelinna** [comment]
After click on the checkbox "Usuwanie instytucji " User see "Nie posiadasz uprawnien do przegladania instytucji"
image

File: s4e-web/src/app/state/session/session.query.ts [hunk 1]
```diff
@@ -34,6 +34,18 @@ export class SessionQuery extends Query<Session> {
   isAdmin() {
     return this.getValue().admin;
   }
+
+  canDeleteInstitution() {
+    return this.getValue().authorities
+      .some(authority => authority === 'OP_INST_DELETE')
```
Comment (michal-szostak) [inline s4e-web/src/app/state/session/session.query.ts:40]: for arrays of strings you can just use `.authorities.includes('OP_INST_DELETE')`

File: s4e-web/src/app/state/session/session.query.ts [hunk 2]
```diff
@@ -34,6 +34,18 @@ export class SessionQuery extends Query<Session> {
...
+      || this.getValue().admin;
+  }
+  canGrantInstitutionDeleteAuthority() {
+    return this.getValue().authorities
+      .some(authority => authority === 'OP_INST_DELETE_GRANT')
```
Comment (michal-szostak) [inline s4e-web/src/app/state/session/session.query.ts:45]: same as above

File: s4e-web/src/app/views/settings/manage-authorities/manage-authorities.component.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    this._personService.fetchAllAdmins$()
+      .pipe(untilDestroyed(this))
+      .subscribe();
+    this.canGrantDeleteAuthority = this._sessionQuery.canGrantInstitutionDeleteAuthority();
+    this._sessionQuery.select('email').pipe(untilDestroyed(this)).subscribe(email => this.currentUserEmail = email);
+  }
+
+  toggleDeleteAuthority(person: Person) {
+    this._personService.toggleDeletePrivilegeFor$(person)
+      .pipe(untilDestroyed(this))
```
Comment (michal-szostak) [inline s4e-web/src/app/views/settings/manage-authorities/manage-authorities.component.ts:38]: When observable is guaranteed to complete (all http calls) `untilDestroyed(this)` is redundant, and also suggests that this observable is long lived

**Review by michal-szostak — APPROVED** [review]

File: s4e-web/src/app/state/session/session.query.ts [hunk 4]
```diff
@@ -34,6 +34,14 @@ export class SessionQuery extends Query<Session> {
   isAdmin() {
     return this.getValue().admin;
   }
+
+  canDeleteInstitution() {
+    return this.getValue().authorities.includes('OP_INST_DELETE') || this.getValue().admin;
+  }
+  canGrantInstitutionDeleteAuthority() {
+    return this.getValue().authorities.includes('OP_INST_DELETE_GRANT') || this.getValue().admin;
```
Comment (jswk) [inline s4e-web/src/app/state/session/session.query.ts:42]: The authority which you want is actually `OP_GRANT_OP_INST_DELETE`

**Comment by Marcelinna** [comment]
Error after delete instytution

File: s4e-web/src/app/state/session/session.query.ts [hunk 5]
```diff
@@ -34,6 +34,14 @@ export class SessionQuery extends Query<Session> {
   isAdmin() {
     return this.getValue().admin;
   }
+
+  canDeleteInstitution() {
+    return this.getValue().authorities.includes('OP_INST_DELETE') || this.getValue().admin;
```
Comment (jswk) [inline s4e-web/src/app/state/session/session.query.ts:39]: The authority should be `OP_INSTITUTION_DELETE`.

**Review by jswk — APPROVED** [review]

**Pull request merged**